implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/DynatraceConfig.java

```diff
@@ -143,6 +146,21 @@ default boolean useDynatraceSummaryInstruments() {
         return getBoolean(this, "useDynatraceSummaryInstruments").orElse(true);
     }
 
+    /**
+     * Toggle whether to export metadata (unit and description) to the Dynatrace backend
+     * for the V2 version of this exporter. Metadata will be exported by default from
+     * Micrometer version 1.12.0. This setting has no effect for the (legacy) Dynatrace
+     * Exporter v1
+     * @return true if metadata should be exported, false otherwise.
+     * @since 1.12.0
+     */
+    default boolean exportMeterMetadata() {
```

Comment:
I don't know exactly what the `enrichWithDynatraceMetadata()` controls, but I wonder if users will be confused by this `exportMeterMetadata()` co-existing with it. What's the motivation to have a configuration method at all for this? Is there a concern of performance issues for users with many meters?

Reply:
`enrichWithDynatraceMetadata()` retrieves Dynatrace-specific metadata from the OneAgent, if it's available. This helps Dynatrace put the Micrometer metrics in the right context in our UI.
`exportMeterMetadata()` controls whether the metadata set on the meters in Micrometer should be exported at all. There is a performance overhead, and since metadata does not change often, users might want to set it once and then turn off this feature. Dynatrace will remember the metadata you set before. We have seen that turning on this feature can increase the volume of exported data significantly, and want to give users (especially ones that export huge volumes of data) an out. Egress can get quite expensive in certain situations, and exporting redundant data does not help that.

I have thought about other ways to tackle this, but those approaches usually have the downside of having to keep state between exports and would be a much bigger refactoring. I might tackle that at a later point in time. We have users asking for the possibility of exporting metadata, and we want to make sure that users who do not want to have the overhead can opt out.

Do you think adding more documentation on these methods would make it more clear? I will also update the Micrometer docs when this feature is added.

Reply:
I added some more context to the Javadocs. I also pointed out that setting that toggle to false is equivalent to configuring a MeterFilter. I think there is value in having it be easy to toggle with a Spring Boot configuration, especially since it might be a somewhat resource-heavy feature.